src/sqlfluff/core/templaters/jinja.py

```diff
@@ -354,15 +354,17 @@ def _slice_template(cls, in_str: str) -> Iterator[RawFileSlice]:
                         trimmed_content = str_buff[
                             len(m_open.group(0)) : -len(m_close.group(0))
                         ]
-                    if trimmed_content.startswith("end"):
```

Comment:
Not related to the PR, but tidies up block types. "directive" means a Jinja command such as `% set elements = "foo" %}`. It is a standalone command, so not a block start, middle, or end.